internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -429,7 +429,7 @@ func (r *Reconciler) createMachineDeployment(ctx context.Context, cluster *clust
 			*ownerReferenceTo(md.Object),
 		})
 		if err := r.reconcileMachineHealthCheck(ctx, nil, md.MachineHealthCheck); err != nil {
-			return errors.Wrapf(err, "failed to create %s", tlog.KObj{Obj: md.MachineHealthCheck})
+			return err
```

Comment:
I think that it is fine to keep the error message here, given that the MHC name is not random anymore at this stage (it is the same name of the machine deployment already created)

Reply:
Kind of agree, but as we're already producing this error inside of `reconcileMachineHealthCheck` I think we don't have to wrap it here again. (so we should just return err)

Reply:
Same in l.469 and l.528 I think then we will propagate the error generated in reconcileMachineHealthCheck on all call sites of the reconcileMachineHealthCheck func